src/mongo/MongoTripodConfig.class.php

```diff
@@ -1152,11 +1319,346 @@ public function getQueueConfig()
+        }
+        if(empty($searchSpecIds))
+        {
+            $searchSpecIds = array_keys($this->searchDocSpecs[$storeName]);
+        }
+        $dataSources = array();
+        foreach($searchSpecIds as $searchSpec)
+        {
+            if(isset($this->searchDocSpecs[$storeName][$searchSpec]))
+            {
+                $dataSources[] = $this->searchDocSpecs[$storeName][$searchSpec]['to'];
+            }
+            else
+            {
+                throw new MongoTripodConfigException("Search document spec id '{$searchSpec}' not in configuration for store '{$storeName}'");
+            }
+        }
+
+        $collections = array();
+        foreach(array_unique($dataSources) as $dataSource)
+        {
+            $collections[] = $this->getMongoCollection(
+                $this->getDatabase($storeName, $dataSource, $readPreference),
+                SEARCH_INDEX_COLLECTION
+            );
+        }
+        return $collections;
+    }
+
+    /**
+     * @param string $storeName
+     * @param string $readPreference
+     * @return MongoCollection
+     */
+    public function getCollectionForTTLCache($storeName, $readPreference = MongoClient::RP_PRIMARY_PREFERRED)
+    {
+        return $this->getMongoCollection(
+            $this->getDatabase($storeName, $this->dbConfig[$storeName]['data_source'], $readPreference),
+            TTL_CACHE_COLLECTION
+        );
+    }
+
+    /**
+     * @param string $storeName
+     * @param string $readPreference
+     * @return MongoCollection
+     */
+    public function getCollectionForLocks($storeName, $readPreference = MongoClient::RP_PRIMARY_PREFERRED)
+    {
+        return $this->getMongoCollection(
+            $this->getDatabase($storeName, $this->dbConfig[$storeName]['data_source'], $readPreference),
+            LOCKS_COLLECTION
+        );
+    }
+
+    /**
+     * @param string $storeName
+     * @param string $readPreference
+     * @return MongoCollection
+     */
+    public function getCollectionForManualRollbackAudit($storeName, $readPreference = MongoClient::RP_PRIMARY_PREFERRED)
+    {
+        return $this->getMongoCollection(
+            $this->getDatabase($storeName, $this->dbConfig[$storeName]['data_source'], $readPreference),
+            AUDIT_MANUAL_ROLLBACKS_COLLECTION
+        );
+    }
+
+    /**
+     * @param $readPreference
+     * @return MongoDB
+     * @throws MongoTripodConfigException
+     */
+    public function getQueueDatabase($readPreference = MongoClient::RP_PRIMARY_PREFERRED)
+    {
+
+        if(!isset($this->dataSources[$this->queueConfig['data_source']]))
+        {
+            throw new MongoTripodConfigException("Data source '" . $this->queueConfig['data_source'] . "' not in configuration");
+        }
+        $connectionOptions = array();
+        $dataSource = $this->dataSources[$this->queueConfig['data_source']];
+        if(isset($dataSource['connectTimeoutMS']) ? $dataSource['connectTimeoutMS'] : 20000);
```

Comment:
Sure.  But there's a bigger issue here.  Like what is this line even trying to achieve...